packages/eds-core-react/src/components/TextField/TextField.tsx

```diff
@@ -82,6 +85,12 @@ export const TextField = forwardRef<HTMLDivElement, TextFieldProps>(
     ref,
   ) {
     const helperTextId = useId(null, 'helpertext')
+    const [, updateState] = useState<object>({})
+    const forceUpdate = useCallback(() => updateState({}), [])
+    //if rightAdornments are updated dynamically, we need to force re-render the component so the Input can set correct padding
+    useEffect(() => {
+      forceUpdate()
+    }, [unit, inputIcon, forceUpdate])
```

Comment:
It would be nice if you could show me where we lose reactivity and why this is needed. I am hoping we can solve this without this complexity. 

In my mind, the component should re-render when the prop changes, and then we recalculate the padding based on the new props.

Reply:
It does re-render when the prop changes and it does recalculate the padding. The problem is this all happens before the dom updates, so it is the old values. In `Input`, this works correctly, but in `TextField`, which wraps `Input` we get the delayed dom update 🤷

Reply:
I understand now. The reason why it was not working as expected for the `TextField` component was that we always passed the `rightAdornments` property. 

I added the `hasRightAdornments` check so that we only pass it when `unit` or `inputIcon` are defined in the props.